### DISCUSSION TITLE
When i am clicking button then open another div in react js

### DISCUSSION THREAD

**Discussion by MinalPatilRPA** [tailwindlabs/tailwindcss#12577] [Help] [answered]
Hi Team

my main page is:- 
fig - main :
![image](https://github.com/tailwindlabs/tailwindcss/assets/122990718/77971542-23d0-4e1b-a74f-78052db30e50)

I want to code like when I click the convert button then after that open another div. as shown in fig below 

![image](https://github.com/tailwindlabs/tailwindcss/assets/122990718/7458b7f5-0d97-4cfe-bffe-d99ebc05bf30)

but my output is like as below fig 2:- 

![image](https://github.com/tailwindlabs/tailwindcss/assets/122990718/bbe85d3c-38fe-45ed-a69c-e76385350b84)

my code is: -

1. Hero.jsx file:-

import React, { useState } from 'react'
import Converter from './Converter';

const Hero = () => {

  const [showDiv, setShowDiv]=useState(false);

  const handleClick = () => {
    setShowDiv(!showDiv);
  };

  return (
    <div className=' p-20 '>
      <div className='text-emerald-500 text-4xl flex flex-row items-center justify-center p-3 '>
        <h1 >
           <span>Online YouTube to MP3 Converter and Downloader</span>
        </h1>
      </div>

      <div className='mx-auto w-1/2 grid grid-rows-2 pt-10 items-center '>
          <h1 className='text-white flex flex-row items-center justify-start text-lg pl-4 '>Please insert a valid You Tube video URL</h1>
        
          <div className='flex flex-row justify-center items-center gap-2 '>
              <input placeholder='https://www.youtube.com/watch?v=doo5ZthfVo6I' className=' p-3 w-[80%] text-black placeholder-gray-400 border-2 border-emerald-500 opacity-60 rounded-full focus:bg-black focus:text-white focus:ring-0 focus:ring-emerald-400 focus:outline-none focus:shadow-md focus:shadow-emerald-300 '/>
              <button className='bg-emerald-400 p-3 w-[20%] text-lg rounded-full hover:bg-emerald-500' onClick={handleClick} >Convert</button>
              {showDiv ? (
                <Converter/>
              ) : null}
             
          </div>
      </div> 
    </div>
  )
}

export default Hero

2. Converter.jsx file:-

import React from 'react'

const Converter = () => {
  return (
   
  <>
      <div className='w-full flex ml-[600px]'>
          <div className=' w-1/4 flex flex-col bg-gray-900 p-4 opacity-80'>
                <div>
                    <span className='text-white text-xl items-center font-semibold'>Audio</span>         
                </div>

                <div className='grid grid-cols-2 items-center gap-36 pb-4 pt-4 font-light'>
                    <span className='text-white text-lg'>MP3</span>
                    <button className=' bg-red-500  py-1 rounded-sm flex justify-center items-center hover:bg-green-700'>
                              <svg className=' text-white' xmlns="http://www.w3.org/2000/svg" width="60" height="25" fill="currentColor" class="bi bi-download" viewBox="0 0 16 16">
              <path d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"></path>
              <path d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"></path>
                              </svg>
                    </button>
                </div>
                  
                <div>
                    <span className='text-white text-xl items-center font-semibold'>Video</span>      
                </div>

                <div className='grid grid-cols-2 items-center gap-36 pb-4 pt-4 font-light'>
                    <span className='text-white text-lg'>1080p (.mp4)</span>
                    <button className=' bg-red-500  py-1 rounded-sm flex justify-center items-center hover:bg-green-700'>
                              <svg className=' text-white' xmlns="http://www.w3.org/2000/svg" width="90" height="25" fill="currentColor" class="bi bi-download" viewBox="0 0 16 16">
              <path d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"></path>
              <path d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"></path>
                              </svg>
                    </button>
                </div>
                
                <div className='grid grid-cols-2 items-center gap-36 pb-4 font-light'>
                    <span className='text-white text-lg'>720p (.mp4)</span>
                    <button className=' bg-red-500  py-1 rounded-sm flex justify-center items-center hover:bg-green-700'>
                              <svg className=' text-white' xmlns="http://www.w3.org/2000/svg" width="90" height="25" fill="currentColor" class="bi bi-download" viewBox="0 0 16 16" >
              <path d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"></path>
              <path d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"></path>
                              </svg>
                    </button>
                </div>

                <div className='grid grid-cols-2 items-center gap-36 pb-4 font-light'>
                    <span className='text-white text-lg'>360p (.mp4) </span>
                    <button className=' bg-red-500  py-1 rounded-sm flex justify-center items-center hover:bg-green-700'>
                              <svg className=' text-white' xmlns="http://www.w3.org/2000/svg" width="90" height="25" fill="currentColor" class="bi bi-download" viewBox="0 0 16 16">
              <path d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"></path>
              <path d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"></path>
                              </svg>
                    </button>
                </div>

                
          </div>

          <div className=' w-1/4 flex flex-col bg-gray-900 p-4 opacity-80'>
              <div class="mt-2">
                  {/* <iframe src="https://www.youtube.com/watch?v=TFX19GQ8LMQ&t=283s" width={'100%'} height={300} frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe> */}
              </div>
          </div>

      </div>
          
  </> 
  )
}

export default Converter

How can I solve this issue?

Thanks 
Minal P

**Comment by wongjn** [comment]
Please consider correcting the format of your code. As it is currently, it is difficult to discern.

**Comment by Zehad-10** [comment, marked as answer]
````
<div>
  <div className="mx-auto w-1/2 grid grid-rows-2 pt-10 items-center ">
    <h1 className="text-white flex flex-row items-center justify-start text-lg pl-4 ">Please insert a valid You Tube video URL</h1>

    <div className="flex flex-row justify-center items-center gap-2 ">
      <input placeholder="https://www.youtube.com/watch?v=doo5ZthfVo6I" className=" p-3 w-[80%] text-black placeholder-gray-400 border-2 border-emerald-500 opacity-60 rounded-full focus:bg-black focus:text-white focus:ring-0 focus:ring-emerald-400 focus:outline-none focus:shadow-md focus:shadow-emerald-300 " />
      <button className="bg-emerald-400 p-3 w-[20%] text-lg rounded-full hover:bg-emerald-500" onClick="{handleClick}">Convert</button>
    </div>
  </div>

  <div>
{showDiv ? (
  <Converter />
  ) : null}
</div>
</div>``

if i understand correctly then try like this take that `{showDiv ? (
  <Converter />
  ) : null}` outside of that flex container
